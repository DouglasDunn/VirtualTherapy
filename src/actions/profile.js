import uuid from 'uuid';
import database from '../firebase/firebase';

// CREATE_PROFILE
export const createProfile = (profile) => ({
  type: 'CREATE_PROFILE',
  profile
});

export const startCreateProfile = (profileData = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    const {
      name = '',
      age = 0,
      emergencyContactName = '',
      emergencyContactNumber = ''
    } = profileData;
    const profile = { name, age, emergencyContactName, emergencyContactNumber };

    return database.ref(`users/${uid}/profile`).set(profile).then((ref) => {
      dispatch(createProfile({
        ...profile
      }));
    });
  };
};

// SET_PROFILE
export const setProfile = (profile) => ({
  type: 'SET_PROFILE',
  profile
});

export const startSetProfile = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/profile`).once('value').then((snapshot) => {
      dispatch(setProfile(snapshot.val()));
    });
  };
};
