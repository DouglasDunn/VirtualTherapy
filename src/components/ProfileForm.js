import React from 'react';

export default class ProfileForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      firstName: props.profile ? props.profile.firstName : '',
      lastName: props.profile ? props.profile.lastName : '',
      emailAddress: props.profile ? props.profile.emailAddress : '',
      dateOfBirth: props.profile ? props.profile.dateOfBirth : '',
      gender: props.profile ? props.profile.gender : 'male',
      emergencyContactName: props.profile ? props.profile.emergencyContactName : '',
      emergencyContactNumber: props.profile ? props.profile.emergencyContactNumber : ''
    };
  }

  onFirstNameChange = (e) => {
    const firstName = e.target.value;
    this.setState(() => ({ firstName }));
  };

  onLastNameChange = (e) => {
    const lastName = e.target.value;
    this.setState(() => ({ lastName }));
  };

  onEmailAddressChange = (e) => {
    const emailAddress = e.target.value;
    this.setState(() => ({ emailAddress }));
  };

  onDateOfBirthChange = (e) => {
    const dateOfBirth = e.target.value;
    this.setState(() => ({ dateOfBirth }));
  };

  onGenderChange = (e) => {
    const gender = e.target.value;
    this.setState(() => ({ gender }));
  };

  onEmergencyContactNameChange = (e) => {
    const emergencyContactName = e.target.value;
    this.setState(() => ({ emergencyContactName }));
  };

  onEmergencyContactNumberChange = (e) => {
    const emergencyContactNumber = e.target.value;
    this.setState(() => ({ emergencyContactNumber }));
  };

  onSubmit = (e) => {
    e.preventDefault();

    this.props.onSubmit({
      firstName: this.state.firstName,
      lastName: this.state.lastName,
      emailAddress: this.state.emailAddress,
      dateOfBirth: this.state.dateOfBirth,
      gender: this.state.gender,
      emergencyContactName: this.state.emergencyContactName,
      emergencyContactNumber: this.state.emergencyContactNumber,
    });
  };

  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        <input
          type="text"
          placeholder="First Name"
          autoFocus
          className="text-input"
          value={this.state.firstName}
          onChange={this.onFirstNameChange}
        />
        <input
          type="text"
          placeholder="Last Name"
          className="text-input"
          value={this.state.lastName}
          onChange={this.onLastNameChange}
        />
        <input
          type="text"
          placeholder="Email Address"
          className="text-input"
          value={this.state.emailAddress}
          onChange={this.onEmailAddressChange}
        />
        <input
          type="text"
          placeholder="Date of Birth"
          className="text-input"
          value={this.state.dateOfBirth}
          onChange={this.onDateOfBirthChange}
        />
        <label>
          <input
            type="radio"
            checked={this.state.gender === 'male'}
            className="radio-input"
            value="male"
            onChange={this.onGenderChange}
          />
          Male
        </label>
        <label>
          <input
            type="radio"
            checked={this.state.gender === 'female'}
            className="radio-input"
            value="female"
            onChange={this.onGenderChange}
          />
          Female
        </label>
        <label>
          <input
            type="radio"
            checked={this.state.gender === 'other'}
            className="radio-input"
            value="other"
            onChange={this.onGenderChange}
          />
          Other
        </label>
        <input
          type="text"
          placeholder="Emergency Contact Name"
          className="text-input"
          value={this.state.emergencyContactName}
          onChange={this.onEmergencyContactNameChange}
        />
        <input
          type="text"
          placeholder="Emergency Contact Number"
          className="text-input"
          value={this.state.emergencyContactNumber}
          onChange={this.onEmergencyContactNumberChange}
        />
        <div>
          <button className="button">Submit</button>
        </div>
      </form>
    )
  }
}
