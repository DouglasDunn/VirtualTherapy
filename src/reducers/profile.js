// Profiles Reducer

const profileReducerDefaultState = {};

export default (state = profileReducerDefaultState, action) => {
  switch (action.type) {
    case 'CREATE_PROFILE':
      return {
        ...action.profile
      };
    case 'SET_PROFILE':
      return action.profile;
    default:
      return state;
  }
};
