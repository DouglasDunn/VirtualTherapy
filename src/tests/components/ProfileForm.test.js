import React from 'react';
import { shallow } from 'enzyme';
import ProfileForm from '../../components/ProfileForm';
import profile from '../fixtures/profile';

test('should render ProfileForm correctly', () => {
  const wrapper = shallow(<ProfileForm />);
  expect(wrapper).toMatchSnapshot();
});

test('should set name on input change', () => {
  const value = 'New name';
  const wrapper = shallow(<ProfileForm />);
  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('name')).toBe(value);
});

test('should set age on input change', () => {
  const value = 'New age';
  const wrapper = shallow(<ProfileForm />);
  wrapper.find('input').at(1).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('age')).toBe(value);
});

test('should set emergencyContactName on input change', () => {
  const value = 'New emergencyContactName';
  const wrapper = shallow(<ProfileForm />);
  wrapper.find('input').at(2).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('emergencyContactName')).toBe(value);
});

test('should set emergencyContactNumber on input change', () => {
  const value = 'New emergencyContactNumber';
  const wrapper = shallow(<ProfileForm />);
  wrapper.find('input').at(3).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('emergencyContactNumber')).toBe(value);
});
