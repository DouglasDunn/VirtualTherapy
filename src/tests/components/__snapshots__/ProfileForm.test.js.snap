// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProfileForm correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Age"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Emergency Contact Name"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Emergency Contact Number"
    type="text"
    value=""
  />
  <div>
    <button
      className="button"
    >
      Submit
    </button>
  </div>
</form>
`;
